### PR TITLE
fix: Address a few obvious issues

### PR DESCRIPTION
**Pull request by orloffm**
This doesn't change any behaviors. It clears a few obvious issues I've encountered while trying to use the repository.

1. Proper links to documents from README.md.
2. Code formatting in two README.md files.
3. Typos, e.g. "reciving".
4. Fixed the `add_body` example and added tests for it.

### CODE REVIEW

File: callouts/python/README.md [hunk 1]
```diff
@@ -2,64 +2,69 @@
...
 The minimal operation of this Python-based ext_proc server requires the `grpcio` python package as well as the protobuf generator tool [buf](https://buf.build/docs/introduction).
 
-
 The prefered method of installation is through a virtual enviornment, `venv`. To set up the virtual enviornment run:
 
-```
+```shell
+cd service-extensions-samples/callouts/python
 python -m venv env
-source env/bin activate
```
Comment (orloffm) [inline callouts/python/README.md:23]: Didn't work.

File: callouts/python/README.md [hunk 2]
```diff
@@ -2,64 +2,69 @@
...
 pip install -r requirements-test.txt
 ```
 
 `buf` can be installed from [here](https://buf.build/docs/installation).
 
-The proto library files are generated with `buf` using: 
+The proto library files are generated with `buf` using:
 
-```
-buf -v generate https://github.com/envoyproxy/envoy.git#subdir=api --path envoy/service/ext_proc/v3/external_processor.proto --include-imports
```
Comment (orloffm) [inline callouts/python/README.md:43]: I've split lines like these.

File: callouts/python/extproc/example/add_body/service_callout_example.py [hunk 3]
```diff
@@ -31,13 +32,13 @@ def on_request_body(
       self, body: service_pb2.HttpBody, context: ServicerContext
   ) -> service_pb2.BodyResponse:
     """Custom processor on the request body."""
-    return callout_server.add_body_mutation(body='-added-body')
+    return callout_tools.add_body_mutation(body='-added-body')
```
Comment (rshriram) [inline callouts/python/extproc/example/add_body/service_callout_example.py:35]: why is this change being done?

**Review by rshriram — APPROVED** [review]

File: callouts/python/extproc/example/add_body/service_callout_example.py (hunk 3 above)
Comment (jstraceski) [reply callouts/python/extproc/example/add_body/service_callout_example.py:35]: It looks like the swap over to callout tools was never completed for this file, this is good, there should have been a test to catch this. Thank you for adding one.

File: callouts/python/README.md (hunk 1 above)
Comment (jstraceski) [reply callouts/python/README.md:23]: Good catch.

File: callouts/python/README.md (hunk 2 above)
Comment (jstraceski) [reply callouts/python/README.md:43]: Looks good to me.

**Review by jstraceski — APPROVED** [review]

**Pull request merged**